### PR TITLE
Feature/yml config

### PR DESCRIPTION
**Pull request by anton-sidelnikov**
#8 - configuration.go for config read/write
#10 - Initialize func in app.go, in case of DEBUG call sqlite entities.db and create table 'entities' if not exist

### CODE REVIEW

File: main/configuration.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	Debug      bool   `yaml:"debug"`
+	ServerPort int    `yaml:"server_port"`
+	PgDbURL    string `yaml:"pg_db_url"`
+	PgDatabase string `yaml:"pg_database"`
+	PgUsername string `yaml:"pg_username"`
+	PgPassword string `yaml:"pg_password"`
+}
+
+// InitConfiguration method for writeConfiguration func
+func (c *Configuration) InitConfiguration(debug bool) Configuration {
```
Comment (outcatcher) [inline main/configuration.go]: If used only in writeConfiguration, should be private

File: main/configuration.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	configuration := Configuration{}
```
Comment (outcatcher) [inline main/configuration.go]: Following can be replaced with constructor

File: main/configuration.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+		configuration.PgDbURL = "localhost:9999"
+		configuration.PgUsername = "entities"
+		configuration.PgPassword = ""
+		configuration.ServerPort = 5054
+	}
+	return configuration
+}
+
+// loadConfiguration load configuration from config.yml
+func (c *Configuration) loadConfiguration() *Configuration {
```
Comment (outcatcher) [inline main/configuration.go]: Config path is not parametrizable

File: main/configuration.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	if _, err := os.Stat(defaultCfgPATH); os.IsNotExist(err) {
+		// write to file
+		f, err := os.Create(defaultCfgPATH)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		err = ioutil.WriteFile(defaultCfgPATH, data, 0644)
+		if err != nil {
+			log.Fatal(err)
```
Comment (outcatcher) [inline main/configuration.go]: This happens during server initialization. Server should be halted, panic expected (in other similar cases too)

File: main/configuration.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+	}
+	err = yaml.Unmarshal(yamlFile, c)
+	if err != nil {
+		log.Fatalf("Unmarshal: %v", err)
+	}
+	return c
+}
+
+// writeConfiguration write config.yml if it not exist with debug or not mode
+func (c *Configuration) writeConfiguration(debug bool) *Configuration {
```
Comment (outcatcher) [inline main/configuration.go]: Should be Public?

File: main/app.go [hunk 6]
```diff
@@ -18,31 +21,46 @@ type App struct {
...
-//InitializeRoutes method
+//InitializeRoutes - init routes for api requests
 func (a *App) InitializeRoutes() {
 	a.Router.HandleFunc("/", a.Ok).Methods("GET")
 	a.Router.HandleFunc("/entities", a.GetEntities).Methods("GET")
 	a.Router.HandleFunc("/entity", a.CreateEntity).Methods("POST")
-	a.Router.HandleFunc("/entity/{id:[0-9]+}", a.GetEntity).Methods("GET")
-	a.Router.HandleFunc("/entity/{id:[0-9]+}", a.UpdateEntity).Methods("PUT")
-	a.Router.HandleFunc("/entity/{id:[0-9]+}", a.DeleteEntity).Methods("DELETE")
+	a.Router.HandleFunc("/entity/{id:[a-z0-9_-]*}", a.GetEntity).Methods("GET")
```
Comment (outcatcher) [inline main/app.go]: Too broad: UUID v4 contains only hex numbers and `-`, also, it has fixed size, 36 chars always

File: main/main.go [hunk 7]
```diff
@@ -1,13 +1,22 @@
...
+	var (
+		config Configuration
+		debug  = true
+	)
+	config.writeConfiguration(debug)
+	config.loadConfiguration()
+	fmt.Printf("Load config\n")
+	a.Initialize(config)
+	fmt.Printf("Init app\n")
+	//os.Getenv("APP_DB_USERNAME"),
```
Comment (outcatcher) [inline main/main.go]: Commented rows should be removed

File: main/app.go [hunk 8]
```diff
@@ -18,31 +21,46 @@ type App struct {
...
-//Initialize method
-func (a *App) Initialize(user, password, dbname string) {
-	connectionString := fmt.Sprintf("user=%s password=%s dbname=%s sslmode=disable", user, password, dbname)
+//Initialize func: init server according configuration structure
+func (a *App) Initialize(config Configuration) {
 	var err error
-	a.DB, err = sql.Open("postgres", connectionString)
-	if err != nil {
-		log.Fatal(err)
+	if !config.Debug {
```
Comment (outcatcher) [inline main/app.go:29]: Maybe move two cases to two functions?

File: config.yml [hunk 9]
```diff
@@ -0,0 +1,6 @@
+debug: true
```
Comment (outcatcher) [inline config.yml]: Is this file used anywhere?

**Comment by outcatcher** [comment]
Complex task. Tests needed, at least for config loading.

**Comment by outcatcher** [comment]
I suppose, entities.db was pushed by mistake?

**Review by outcatcher — CHANGES_REQUESTED** [review]

File: main/configuration.go (hunk 1 above)
Comment (anton-sidelnikov) [reply main/configuration.go]: ok, will do it

File: main/configuration.go (hunk 2 above)
Comment (anton-sidelnikov) [reply main/configuration.go]: это в принципе он и есть, но немного переделал

File: main/configuration.go (hunk 1 above)
Comment (anton-sidelnikov) [reply main/configuration.go]: ok

File: main/configuration.go (hunk 5 above)
Comment (anton-sidelnikov) [reply main/configuration.go]: no, this is class method

File: config.yml (hunk 9 above)
Comment (anton-sidelnikov) [reply config.yml]: pushed by mistake, will remove it

File: main/app.go (hunk 8 above)
Comment (anton-sidelnikov) [reply main/app.go:29]: maybe, will think about it..

Comment (anton-sidelnikov) [reply main/app.go:29]: so i doesn't like how it looks when i split this to two func

File: main/app.go [hunk 10]
```diff
@@ -18,31 +21,47 @@ type App struct {
...
-//Run method
+//Run server
 func (a *App) Run(addr string) {
 	log.Fatal(http.ListenAndServe(addr, a.Router))
 }
 
-//InitializeRoutes method
+//InitializeRoutes - init routes for api requests
 func (a *App) InitializeRoutes() {
+	var routeUUID4 = "/entity/{id:[a-z0-9]{8}-[a-z0-9]{4}-[1-5][a-z0-9]{3}-[a-z0-9]{4}-[a-z0-9]{12}}"
```
Comment (outcatcher) [inline main/app.go]: Can be made a const

**Review by outcatcher — COMMENTED** [review]
Need review from @anton-sidelnikov

**Comment by anton-sidelnikov** [comment]
all seems ok, now

**Review by outcatcher — APPROVED** [review]

**Pull request merged**